packages/zero-solid/src/use-zero.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  batch,
+  createContext,
+  createMemo,
+  onCleanup,
+  useContext,
+  type JSX,
+} from 'solid-js';
+import {
+  Zero,
+  type CustomMutatorDefs,
+  type Schema,
+  type ZeroOptions,
+} from '../../zero/src/zero.ts';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention, @typescript-eslint/no-explicit-any
+const ZeroContext = createContext<(() => Zero<any, any>) | undefined>(
+  undefined,
+);
+
+export function createZero<S extends Schema, MD extends CustomMutatorDefs<S>>(
+  options: ZeroOptions<S, MD>,
+): Zero<S, MD> {
+  const opts = {
+    ...options,
+    batchViewUpdates: batch,
+  };
+  return new Zero(opts);
+}
+
+export function useZero<
+  S extends Schema,
+  MD extends CustomMutatorDefs<S> | undefined = undefined,
+>(): () => Zero<S, MD> | undefined {
+  const zero = useContext(ZeroContext);
+
+  // TODO: uncomment when we require ZeroProvider in a future release.
+  // if (zero === undefined) {
+  //   throw new Error('useZero must be used within a ZeroProvider');
+  // }
+  return zero ?? (() => undefined);
+}
+
+export function createUseZero<
+  S extends Schema,
+  MD extends CustomMutatorDefs<S> | undefined = undefined,
+>() {
+  return () => useZero<S, MD>();
+}
+
+export function ZeroProvider<
+  S extends Schema,
+  MD extends CustomMutatorDefs<S> | undefined = undefined,
+>(props: {
+  children: JSX.Element;
+  zeroSignal: () => ZeroOptions<S, MD> | {zero: Zero<S, MD>};
```

Comment:
I find this type a bit strange. I want the props to be called zero and options.

Is this how the React one is done?

Reply:
I fixed this to more closely match the React API.  So its either:

```
<ZeroProvider zero={zero}>...</ZeroProvider>
```

or

```
<ZeroProvider server="https://foo.com" userID="u1" schema={schema}>...</ZeroProvider>
```